### PR TITLE
bears/dart: Add bear for Dart using Dart Linter

### PR DESCRIPTION
**Pull request by Adrianzatreanu**
Add a bear for Dart, which is a scalable programming language.
It looks a bit like a python or javascript.

Fixes https://github.com/coala-analyzer/coala-bears/issues/105

### CODE REVIEW

**Comment by sils1297** [comment]
why close it? Just fix it and force puh

**Comment by sils1297** [comment]
puh -> push

**Comment by Adrianzatreanu** [comment]
phew, seems to be fine now :)

**Comment by sils1297** [comment]
you can just run coala to let it fix your code automatically, you know that? GitMate uses coala.

**Comment by Adrianzatreanu** [comment]
no, i actually didnt know, its quite interesting :) however, it was small bugfixes so its ok :D

**Comment by Adrianzatreanu** [comment]
99,81, so close, yet so far ! :(

File: bears/dart/DartLintBear.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    severity_map = {
+        "[error]": RESULT_SEVERITY.MAJOR,
+        "[warning]": RESULT_SEVERITY.NORMAL}
+
+    def run(self, filename, file):
+        '''
+        Checks the code with `dart-linter`. Note: This bear expects dartanalyzer
+        to be on your PATH. Please ensure /path/to/dart-sdk/bin is in your PATH.
+        '''
+        self.arguments = ' {filename}'
```
Comment (SanketDG) [inline bears/dart/DartLintBear.py]: You do not need this space, there is already a space inserted.

File: bears/dart/DartLintBear.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        r'(?P<file_name>.+),\s[l][i][n][e]\s(?P<line>.*),'
+        r'\s[c][o][l]\s(?P<column>.*)\)')
+    severity_map = {
+        "[error]": RESULT_SEVERITY.MAJOR,
+        "[warning]": RESULT_SEVERITY.NORMAL}
+
+    def run(self, filename, file):
+        '''
+        Checks the code with `dart-linter`. Note: This bear expects dartanalyzer
+        to be on your PATH. Please ensure /path/to/dart-sdk/bin is in your PATH.
```
Comment (SanketDG) [inline bears/dart/DartLintBear.py]: Can you please say `$PATH`

Comment (SanketDG) [inline bears/dart/DartLintBear.py]: And make system paths and `$PATH` as code blocks.

File: bears/dart/DartLintBear.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+from coalib.bearlib.abstractions.Lint import Lint
+from coalib.bears.LocalBear import LocalBear
+from coalib.results.RESULT_SEVERITY import RESULT_SEVERITY
+
+
+class DartLintBear(LocalBear, Lint):
+    executable = 'dartanalyzer'
+    output_regex = re.compile(
+        r'(?P<severity>\[error\]|\[warning\])\s(?P<message>.*)\('
+        r'(?P<file_name>.+),\s[l][i][n][e]\s(?P<line>.*),'
```
Comment (SanketDG) [inline bears/dart/DartLintBear.py]: why not just

```
r'(?P<file_name>.+),\sline\s(?P<line>.*),'
```

Would that not work?

File: bears/dart/DartLintBear.py (hunk 2 above)
Comment (Adrianzatreanu) [inline bears/dart/DartLintBear.py]: sorry, i dont know what u mean by making them "as code blocks"

File: bears/dart/DartLintBear.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        r'\s[c][o][l]\s(?P<column>.*)\)')
+    severity_map = {
+        "[error]": RESULT_SEVERITY.MAJOR,
+        "[warning]": RESULT_SEVERITY.NORMAL}
+
+    def run(self, filename, file):
+        '''
+        Checks the code with `dart-linter`. Note: This bear expects dartanalyzer
```
Comment (sils1297) [inline bears/dart/DartLintBear.py]: please use

    ``code``

as this is rst

File: bears/dart/DartLintBear.py (hunk 2 above)
Comment (sils1297) [inline bears/dart/DartLintBear.py]: just enclose it in those:

    ``PATH``

(I don't think the dollar is needed, that's bashy)

**Comment by Adrianzatreanu** [comment]
i hope this is better :)

**Comment by sils1297** [comment]
ack 4a1643ae0f2d738

**Comment by sils1297** [comment]
ack 93b3d8f

**Comment by Redridge** [comment]
@rultor merge

**Pull request merged**